File: OLD_FURGOCASA_DATOS/parse-orders.js
// Script para parsear reservas del dump SQL
const fs = require('fs');
const path = require('path');

console.log('📖 Leyendo archivo de reservas...');

// Leer el archivo con el INSERT statement
const rawData = fs.readFileSync(path.join(__dirname, 'orders-raw.txt'), 'utf-8');

// Extraer solo la parte de VALUES
const match = rawData.match(/INSERT INTO `fur_vikrentcar_orders` VALUES (.+);/s);
if (!match) {
    console.error('❌ No se encontró el INSERT statement');
    process.exit(1);
}

let valuesStr = match[1];

console.log(`📊 Parseando reservas (esto puede tardar)...`);

// Las filas están separadas por ),(
// Dividir cuidadosamente
const rows = [];
let current = '';
let inString = false;
let parenCount = 0;

for (let i = 0; i < valuesStr.length; i++) {
    const char = valuesStr[i];
    const prevChar = i > 0 ? valuesStr[i - 1] : '';
    
    // Toggle string state
    if (char === "'" && prevChar !== '\\') {
        inString = !inString;
    }
    
    if (!inString) {
        if (char === '(') parenCount++;
        if (char === ')') parenCount--;
        
        // Si encontramos ),( y estamos en nivel 0, es un separador
        if (char === ',' && parenCount === 0 && prevChar === ')') {
            rows.push(current);
            current = '';
            continue;
        }
    }
    
    current += char;
}

// Agregar última fila
if (current) {
    rows.push(current);
}

console.log(`✅ ${rows.length} reservas encontradas en total`);

// Parsear cada fila
const allBookings = [];
const now = Math.floor(Date.now() / 1000);

for (const rowStr of rows) {
    try {
        // Limpiar paréntesis
        let cleaned = rowStr.trim();
        if (cleaned.startsWith('(')) cleaned = cleaned.substring(1);
        if (cleaned.endsWith(')')) cleaned = cleaned.substring(0, cleaned.length - 1);
        
        // Parsear valores
        const values = [];
        let current = '';
        let inString = false;
        let escaped = false;
        
        for (const char of cleaned) {
            if (escaped) {
                current += char;
                escaped = false;
                continue;
            }
            
            if (char === '\\') {
                escaped = true;
                continue;
            }
            
            if (char === "'" && !escaped) {
                inString = !inString;
                continue;
            }
            
            if (char === ',' && !inString) {
                const val = current.trim();
                if (val === 'NULL' || val === '') {
                    values.push(null);
                } else if (/^-?\d+(\.\d+)?$/.test(val)) {
                    values.push(val.includes('.') ? parseFloat(val) : parseInt(val));
                } else {
                    values.push(val);
                }
                current = '';
                continue;
            }
            
            current += char;
        }
        
        // Agregar último valor
        if (current) {
            const val = current.trim();
            if (val === 'NULL' || val === '') {
                values.push(null);
            } else if (/^-?\d+(\.\d+)?$/.test(val)) {
                values.push(val.includes('.') ? parseFloat(val) : parseInt(val));
            } else {
                values.push(val);
            }
        }
        
        if (values.length < 30) {
            continue; // Fila inválida
        }
        
        // Extraer timestamp de ritiro y consegna
        const ritiro = values[6];
        const consegna = values[8];
        
        const booking = {
            id: values[0],
            ts: values[3],
            status: values[4] || 'confirmed',
            nominative: values[26] || '',
            custmail: values[10] || '',
            phone: values[25] || '',
            country: values[23] || 'ESP',
            idcar: values[5],
            vehicle_name: 'Unknown',
            ritiro: ritiro,
            consegna: consegna,
            days: values[7],
            order_total: values[19] || 0,
            totpaid: values[15] || 0,
            locationvat: values[20],
            adminnotes: values[27],
            optionals: values[11],
            coupon: values[18],
            idplace: values[12] || 1,
            idreturnplace: values[13] || 1,
        };
        
        allBookings.push(booking);
    } catch (error) {
        // Ignorar filas con errores
    }
}

console.log(`✅ ${allBookings.length} reservas parseadas correctamente`);

// Filtrar solo activas
const activeBookings = allBookings.filter(b => b.consegna && b.consegna >= now);
const pendingBookings = activeBookings.filter(b => b.ritiro > now);
const inProgressBookings = activeBookings.filter(b => b.ritiro <= now && b.consegna >= now);

console.log(`\n📊 Estadísticas:`);
console.log(`   Total reservas: ${allBookings.length}`);
console.log(`   Reservas activas/futuras: ${activeBookings.length}`);
console.log(`   - Pendientes (futuras): ${pendingBookings.length}`);
console.log(`   - En curso: ${inProgressBookings.length}`);

// Guardar archivos
fs.writeFileSync(
    path.join(__dirname, 'all-bookings.json'),
    JSON.stringify(allBookings, null, 2)
);

fs.writeFileSync(
    path.join(__dirname, 'bookings-activas.json'),
    JSON.stringify(activeBookings, null, 2)
);

console.log(`\n✅ Archivos guardados:`);
console.log(`   - all-bookings.json`);
console.log(`   - bookings-activas.json`);
console.log(`\n🚀 Ahora ejecuta: npm run migrate:import`);

File: OLD_FURGOCASA_DATOS/parse-orders.py
import re
import json
from datetime import datetime

# Leer el archivo con el INSERT statement
with open('orders-raw.txt', 'r', encoding='utf-8') as f:
    line = f.read()

# Extraer solo la parte de VALUES
match = re.search(r'INSERT INTO `fur_vikrentcar_orders` VALUES (.+);', line, re.DOTALL)
if not match:
    print("No se encontró el INSERT statement")
    exit(1)

values_str = match.group(1)

# Parsear las filas (están separadas por ),(
# Primero, reemplazar ),( por un marcador especial
values_str = values_str.replace('),(', ')|SPLIT|(')

# Dividir por el marcador
rows_str = values_str.split('|SPLIT|')

print(f"Total de filas encontradas: {len(rows_str)}")

all_bookings = []
now_timestamp = int(datetime.now().timestamp())

for i, row_str in enumerate(rows_str):
    # Limpiar paréntesis
    row_str = row_str.strip()
    if row_str.startswith('('):
        row_str = row_str[1:]
    if row_str.endswith(')'):
        row_str = row_str[:-1]
    
    # Parsear valores manualmente
    values = []
    current = ''
    in_string = False
    escaped = False
    
    for char in row_str:
        if escaped:
            current += char
            escaped = False
            continue
        
        if char == '\\':
            escaped = True
            continue
        
        if char == "'" and not escaped:
            in_string = not in_string
            continue
        
        if char == ',' and not in_string:
            val = current.strip()
            if val == 'NULL':
                values.append(None)
            elif val in ('true', 'false'):
                values.append(val == 'true')
            elif re.match(r'^-?\d+(\.\d+)?$', val):
                values.append(float(val) if '.' in val else int(val))
            else:
                values.append(val)
            current = ''
            continue
        
        current += char
    
    # Agregar último valor
    if current:
        val = current.strip()
        if val == 'NULL':
            values.append(None)
        elif val in ('true', 'false'):
            values.append(val == 'true')
        elif re.match(r'^-?\d+(\.\d+)?$', val):
            values.append(float(val) if '.' in val else int(val))
        else:
            values.append(val)
    
    # Verificar si tiene suficientes valores
    if len(values) < 40:
        print(f"Fila {i} tiene solo {len(values)} valores, saltando...")
        continue
    
    # Extraer datos de la reserva
    booking = {
        'id': values[0],
        'ts': values[3],
        'status': values[4],
        'nominative': values[26] if values[26] else '',
        'custmail': values[10] if values[10] else '',
        'phone': values[25] if values[25] else '',
        'country': values[23] if values[23] else 'ESP',
        'idcar': values[5],
        'vehicle_name': 'Unknown',  # Se completará después
        'ritiro': values[6],
        'consegna': values[8],
        'days': values[7],
        'order_total': float(values[19]) if values[19] else 0.0,
        'totpaid': float(values[15]) if values[15] else 0.0,
        'locationvat': float(values[20]) if values[20] else None,
        'adminnotes': values[27],
        'optionals': values[11],
        'coupon': values[18],
        'idplace': values[12],
        'idreturnplace': values[13],
    }
    
    all_bookings.append(booking)

print(f"\nTotal reservas parseadas: {len(all_bookings)}")

# Filtrar solo activas (fecha de devolución >= ahora)
active_bookings = [b for b in all_bookings if b['consegna'] and b['consegna'] >= now_timestamp]

print(f"Reservas activas/futuras: {len(active_bookings)}")

# Guardar todas las reservas
with open('all-bookings.json', 'w', encoding='utf-8') as f:
    json.dump(all_bookings, f, indent=2, ensure_ascii=False)

# Guardar solo activas
with open('bookings-activas.json', 'w', encoding='utf-8') as f:
    json.dump(active_bookings, f, indent=2, ensure_ascii=False)

print(f"\n✅ Archivos creados:")
print(f"   - all-bookings.json ({len(all_bookings)} reservas)")
print(f"   - bookings-activas.json ({len(active_bookings)} reservas)")

# Mostrar algunas estadísticas
if active_bookings:
    print(f"\n📊 Estadísticas de reservas activas:")
    pending = [b for b in active_bookings if b['ritiro'] > now_timestamp]
    in_progress = [b for b in active_bookings if b['ritiro'] <= now_timestamp < b['consegna']]
    print(f"   - Pendientes (futuras): {len(pending)}")
    print(f"   - En curso: {len(in_progress)}")
